### PR TITLE
Add output defs for edit_distance kernel

### PR DESCRIPTION
**Pull request by jinyouzhi**
### PR types

Others

### PR changes

Others

### Describe

Task 17 in #51292

### CODE REVIEW

**Comment by jinyouzhi** [comment]
Could you give some comments? @From00

File: paddle/phi/kernels/cpu/edit_distance_kernel.cc [hunk 1]
```diff
@@ -121,4 +121,6 @@ void EditDistanceKernel(const Context& ctx,
 }  // namespace phi
 
 PD_REGISTER_KERNEL(
-    edit_distance, CPU, ALL_LAYOUT, phi::EditDistanceKernel, float) {}
+    edit_distance, CPU, ALL_LAYOUT, phi::EditDistanceKernel, float) {
+  kernel->OutputAt(1).SetDataType(phi::DataType::INT64);
```
Comment (From00) [inline paddle/phi/kernels/cpu/edit_distance_kernel.cc]: 输出参数编号从0开始，`sequencenum`为第0个参数

Comment (jinyouzhi) [reply paddle/phi/kernels/cpu/edit_distance_kernel.cc]: https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/fluid/layers/edit_distance_cn.html#edit-distance
文档里看起来是第二个

Comment (From00) [reply paddle/phi/kernels/cpu/edit_distance_kernel.cc]: 文档是描述python端的API实现，这个算子python API的参数顺序与C++ kernel不一致，注册输出类型应以C++端的顺序为准。

Comment (jinyouzhi) [reply paddle/phi/kernels/cpu/edit_distance_kernel.cc]: fix

**Review by From00 — APPROVED** [review]
LGTM

**Pull request merged**